### PR TITLE
EES-4716 Part 4 - Add stub data set query GET endpoint, Parquet path resolver and restructured locations meta

### PR DESCRIPTION
**Pull request by ntsim**
This PR:

- Adds a stub for the data query GET endpoint (`QueryDataSetGet`) so integration test writing can commence (EES-5015)
- Adds `ParquetPathResolver` service that can allow us to resolve paths to Parquet files based on the current environment (and any configured option).
- Restructures the locations Parquet meta into a single location table. This is necessary to simplify data set queries as having different tables was found to over-complicate things when processing the queries in the service layer.
- Refactors the seeder to use Parquet table refs to make refactoring column names easier in the future.

Note that the `ParquetPathResolver` hasn't necessarily been completed for integration tests yet. This has been left for the testing ticket to finalise (EES-5015).

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Services/Options/ParquetFilesOptions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using System.ComponentModel.DataAnnotations;
```
Comment (jack-hive) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Services/Options/ParquetFilesOptions.cs:1]: Can remove this `using`

**Review by jack-hive — CHANGES_REQUESTED** [review]

**Review by jack-hive — CHANGES_REQUESTED** [review]

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Parquet/Tables/DataTable.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        public static string LocationId(GeographicLevel geographicLevel)
+            => $"locations_{geographicLevel.GetEnumValue().ToLower()}_id";
+
+        public static string Filter(FilterMeta filter) => $"\"{filter.PublicId}\"";
+
+        public static string Indicator(IndicatorMeta indicator) => $"\"{indicator.PublicId}\"";
+    }
+
+    public static readonly TableRef DefaultRef = new(TableName);
```
[the remark this answers is not in the document]
Comment (jack-hive) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Parquet/Tables/DataTable.cs:24]: On revisiting this, I'm happy with it being left `public`. Just found it weird that we have two ways to access the same thing at first.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Controllers/DataSetsController.cs [hunk 3]
```diff
@@ -107,11 +102,156 @@ public DataSetsController(IDataSetService dataSetService)
...
+    ///
+    /// - `2020|AY` is the 2020/21 academic year
+    /// - `2021|FY` is the 2021/22 financial year
+    /// - `2020|T3` is the 2020/21 academic year's summer term
+    /// - `2020|P2` is the 2020/21 financial year part 2 (October to March)
+    /// - `2020|CY` is the 2020 calendar year
+    /// - `2020|W32` is 2020 week 32
+    /// - `2020/2021|AY` is the 2020/21 academic year
+    /// - `2021/2022|FY` is the 2021/22 financial year
+    ///
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Controllers/DataSetsController.cs:224]: I've adjusted the preceding paragraph which specifies this caveat.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/ViewModels/DataSetQueryResultViewModels.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    [JsonConverter(typeof(EnumToEnumValueJsonConverter<GeographicLevel>))]
+    public required GeographicLevel GeographicLevel { get; init; }
+
+    /// <summary>
+    /// The locations relevant to this result's data.
+    ///
+    /// This is a dictionary where the key is the location's geographic
+    /// level and the value is the location's ID.
+    /// </summary>
+    public required Dictionary<string, string> Locations { get; init; }
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/ViewModels/DataSetQueryResultViewModels.cs:43]: Nope, the locations will be in different levels as rows look like this:

country_name | country_code | region_name | region_code | the_rest_of_the_data 
-------------|--------------|-------------| ----------- | --------------------
England      | E00000001    | Yorkshire   | E00000002   | ..... 

So the corresponding JSON result would need to look something like this:

```json
{
  "locations": {
     "NAT": "id of England"
     "REG": "id of Yorkshire"
  }
}
```

Where the different dictionary values will be different depending on where the data was colelcted.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Parquet/Tables/DataTable.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Parquet/Tables/DataTable.cs:24]: I've just changed them all to private as the default refs don't need to be exposed really

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/ViewModels/DataSetQueryResultViewModels.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (jack-hive) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/ViewModels/DataSetQueryResultViewModels.cs:40]: Sorry, I understand this now. Makes sense! At first glance it always messes with my thought process when I see geographic level appearing in multiple places

**Review by jack-hive — APPROVED** [review]
Nice!

**Pull request merged**